app/src/main/java/app/wefridge/wefridge/model/ItemController.kt

```diff
@@ -104,19 +102,20 @@ class ItemController {
         }
 
         private fun parse(item: DocumentSnapshot): Item {
+
             with(item) {
                 return Item(id,
-                    getString(ITEM_NAME) ?: "",
-                    getString(ITEM_DESCRIPTION),
-                    getBoolean(ITEM_IS_SHARED) ?: false,
-                    getLong(ITEM_QUANTITY) ?: 0,
-                    Unit.getByValue(getLong(ITEM_UNIT)?.toInt()) ?: Unit.PIECE,
-                    getTimestamp(ITEM_BEST_BY)?.toDate(),
-                    getGeoPoint(ITEM_LOCATION),
-                    getString(ITEM_GEOHASH),
-                    getString(ITEM_CONTACT_NAME),
-                    getString(ITEM_CONTACT_EMAIL),
-                    getDocumentReference(ITEM_OWNER)
+                    get(ITEM_NAME) as? String ?: "",
+                    get(ITEM_DESCRIPTION) as? String,
+                    get(ITEM_IS_SHARED) as? Boolean ?: false,
+                    get(ITEM_QUANTITY) as? Long ?: 0,
+                    Unit.getByValue((item.get(ITEM_UNIT) as? Long)?.toInt()) ?: Unit.PIECE,
+                    (get(ITEM_BEST_BY) as? Timestamp)?.toDate(),
+                    get(ITEM_LOCATION) as? @RawValue GeoPoint,
+                    get(ITEM_GEOHASH) as? String,
+                    get(ITEM_CONTACT_NAME) as? String,
+                    get(ITEM_CONTACT_EMAIL) as? String,
+                    get(ITEM_OWNER) as? @RawValue DocumentReference
```

Comment:
Why? `getString` checks for the correct type and throws a RuntimeException on wrong type.
See [here](https://github.com/googleapis/java-firestore/blob/main/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentSnapshot.java#L314-L324).

Reply:
We could catch the runtime exception, too. But I found it better to assign null to the values rather than setting the whole object to null if a parameter has a wrong data type.

Reply:
yea ok, makes sense